client/src/services/activity.ts

```diff
@@ -41,3 +42,55 @@ export const fetchPCStatus = async () => {
     return error;
   }
 };
+
+export const getRecentActivity = async () => {
+  const url = `${API_URL}/activity/all/recent`;
+  const settings = {
+    method: "GET",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  try {
+    const response = await fetch(url, settings);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    return error;
+  }
+};
+
+export const useFetchActivities = () => {
+  const store = useBoundStore.getState();
+
+  useEffect(() => {
+    const fetchActivities = async () => {
+      try {
+        const activities = await getRecentActivity();
+
+        const activitiesWithProfiles = await Promise.all(
+          activities.map(async (activity) => {
+            const profile = await getGamerProfile(activity.student_number);
+
+            return {
+              first_name: profile.first_name,
+              last_name: profile.last_name,
+              ...activity,
+            };
+          }),
+        );
+
+        store.setLogList(activitiesWithProfiles);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchActivities();
+
+    const intervalId = setInterval(fetchActivities, 5000); // fetch every 5 seconds
+    return () => clearInterval(intervalId);
+  }, []);
```

Comment:
> I am not sure how to update it based on state instead

Instead of polling, we can just think of the times when we need to update the activity log:
1. On app load
2. When a user checks in

For on app load, calling `useEffect()` with what you did here is great and this should just be one API call without the interval. 

For when a user checks in, calling `fetchActivities()` (line 69) when we check in a user [here](https://github.com/ubcesports/lounge-hub/blob/042bddb9859feb8012ca979bd09c4179267f46c9/client/src/app/check-in/check-in.tsx#L25C34-L25C39) should update it because we are updating store state which re-renders the component.